dcpy/utils/data.py

```diff
@@ -0,0 +1,156 @@
+import pandas as pd
+import geopandas as gpd
+
+from pathlib import Path
+
+from dcpy.utils.logging import logger
+from dcpy.models import file
+from dcpy.models.lifecycle.ingest import (
+    Config,
+)
+
+import zipfile
+
+EXTACTED_FILES_DIR = Path("extracted_files")
+
+
+def read_data_to_df(
+    config: Config, local_data_path: Path
```

Comment:
If this function is going to take a Config (or any model from `lifecycle`), I don't think it should be in utils. Sort of goes with our discussion today. I know you wanted to limit scope, so for now, maybe it could just take a file.Format and a filename

Reply:
Oops you're already doing that. Seems like you can just pass in the file format. For the errors that refer to the raw file name, you can do `local_data_path.stem`

Reply:
oops not stem. `local_data_path.name`

Reply:
This function will be taking in a model from `lifecycle` - where would you recommend placing this fn in such case? `dcpy.lifeclycle.ingest.tranform`? 

This `data.py` file also has `unzip_file` fn below. Lmk if it makes sense to keep in utils, `data.py` file.

Reply:
* Regarding passing in the file format: if it's okay, I would like to hold off this change until next PR because my test also uses this fn. The reorganization of the test code, creating input objects for the `test_to_parquet` fn, will be a bit messy.

* Regarding raw file name, I don't need-need it in this function per se. The only places I'm referring to it are assert messages below: 
    <img width="867" alt="image" src="https://github.com/NYCPlanning/data-engineering/assets/144725249/b8363e23-08d4-4189-a35e-848d0774c7ab">

   I can def replace it with the Path.name attribute - ty for the suggestion! The only thing is when an input path is zipped csv, the assert message will say "... doesn't exist in `file.csv.zip`" which is fine.

Reply:
If you switched to a `file.Format` object as the arg instead of config, it should just be a one-line difference in the tests, no? There'd be room to make things cleaner by not goofing the rest of the template like we'd talked about but for now you could just edit this line

https://github.com/NYCPlanning/data-engineering/pull/729/files#diff-888d97f8d2b095ef22f1837f5039b12008ef5ce31d310b5833051877bcd63eccR96

from this
```
raw_df = data.read_data_to_df(config=config, local_data_path=file_path)
```

to this
```
raw_df = data.read_data_to_df(config=config.format, local_data_path=file_path)
```

I think?

Reply:
And then if it's using a more generic type (like `file.Format`, which lives outside of `lifecycle.ingest`) rather than an object very specific to our ingestion code, I think it's fine for this code to keep living in a more pure util file like it is now

Reply:
gotcha! This makes sense. I thought I would need to fix test config objects.